xbmc/utils/StringUtils.cpp

```diff
@@ -759,6 +759,61 @@ std::vector<std::string> StringUtils::Split(const std::string& input, const char
   return results;
 }
 
+std::vector<std::string> StringUtils::Split(const std::vector<std::string> &input, const std::vector<std::string> &delimiters, unsigned int iMaxStrings /* = 0 */)
+{
+  std::vector<std::string> results;
+  if (input.empty())
+    return results;
+
+  for (std::vector<std::string>::const_iterator it = input.begin(); it != input.end(); ++it)
```

Comment:
it's the same thing but imo it's cleaner to use range based iterators to iterate in vectors like this:
```for (const auto& item: input)
  results.push_back(item);
```